bdns/dns.go

```diff
@@ -342,7 +341,7 @@ func (dnsClient *DNSClientImpl) LookupTXT(ctx context.Context, hostname string)
 	for _, answer := range r.Answer {
 		if answer.Header().Rrtype == dnsType {
 			if txtRec, ok := answer.(*dns.TXT); ok {
-				txt = append(txt, strings.Join(txtRec.Txt, ""))
+				txt = append(txt, txtRec.Txt...)
```

Comment:
This goes against RFC 4408 Section 3.1.3 (`If a published record contains multiple strings, then the record MUST be treated as if those strings are concatenated together without adding spaces.`). From the PR desc it sounds like you are talking about _multiple  records_ not multiple strings in a single record, which is what this deals with?

Reply:
I think you're right, which means this PR is obsolete and I'll close it. I think I was misled by a mistake in implementing multiple records in `dns-test-srv`, which I've fixed in c7bca03 (part of #3315).